Modules/binascii.c

```diff
@@ -1130,20 +1130,18 @@ binascii_hexlify_impl(PyObject *module, Py_buffer *data)
     return _Py_strhex_bytes((const char *)data->buf, data->len);
 }
 
-static int
-to_int(int c)
-{
-    if (Py_ISDIGIT(c))
-        return c - '0';
-    else {
-        if (Py_ISUPPER(c))
-            c = Py_TOLOWER(c);
-        if (c >= 'a' && c <= 'f')
-            return c - 'a' + 10;
-    }
-    return -1;
-}
+static const int table_hex[128] = {
```

Comment:
Py_CHARMASK returns values in the range [0, 255], so this table needs to be 256 bytes long. See for example _Py_ctype_tolower.